### PR TITLE
ramda: Allow currying of type guards

### PR DESCRIPTION
**Pull request by sbking**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: Relevant issue in typescript repo: https://github.com/Microsoft/TypeScript/issues/16410
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/ramda/ramda-tests.ts [hunk 1]
```diff
@@ -87,6 +87,25 @@ class F2 {
...
+    interface FastCar { speed: number; }
+
+    let typeGuard = function(a: number, b: number, c: number, d: number, e: number, car: Car): car is FastCar {
+      return car.speed !== undefined;
+    }
+    let typeGuardCurried = R.curry(typeGuard);
+
+    let drive = function(fastCar: FastCar) {};
+
+    let cars: Car[] = [{ speed: 65 }, {}];
```
Comment (mrdziuban) [inline types/ramda/ramda-tests.ts]: You could use `const` for all of these variables since they remain unchanged

**Comment by sbking** [comment]
@mrdziuban Thanks, fixed and squashed!

**Comment by mrdziuban** [comment]
Thanks! LGTM, though I'm not sure what the policy is on merging since I don't have write access.

**Comment by sbking** [comment]
@mrdziuban I'm not sure if you're supposed to thumbs up the original PR comment or the dt-bot comment, but I'm pretty sure once those are thumbs up'd it flags the PR for merging.

**Review by mrdziuban — APPROVED** [review]
👍

**Comment by mrdziuban** [comment]
Okay, I did all of the above, plus approved it, hopefully it amounts to something

**Comment by mdekrey** [comment]
👍 to me too.

**Pull request merged**